File: backend/components/metiers/threed/utils/rbac/core/rbac_core.py
# rbac_core.py
"""Logique métier principale RBAC (Python)"""
def check_permission(user, permission):
    return permission in getattr(user, 'permissions', [])

def get_user_roles(user):
    return getattr(user, 'roles', [])

def has_permission(user, permission):
    # Permissions ultra basiques pour tests (à adapter selon logique métier)
    roles = user.get('roles', [])
    if 'admin' in roles:
        return True
    if 'guest' in roles and permission == 'read':
        return True
    return False

File: backend/components/metiers/threed/utils/rbac/core/rbac_core.js
// rbac_core.js
// Logique métier principale RBAC (JS)
function checkPermission(user, permission) {
  return user.permissions && user.permissions.includes(permission);
}
function getUserRoles(user) {
  return user.roles || [];
}
module.exports = { checkPermission, getUserRoles };
